src/api/app/models/project.rb

```diff
@@ -927,45 +927,66 @@ def add_repository_with_targets(repoName, source_repo, add_target_repos = [], op
-    skip_repos = []
-    a = project.find_attribute('OBS', 'BranchSkipRepositories')
-    skip_repos = a.values.map(&:value) if a
-    project.repositories.each do |repo|
-      next if skip_repos.include? repo.name
-      repoName = opts[:extend_names] ? repo.extended_name : repo.name
-      next if repo.is_local_channel?
-      pkg_to_enable.enable_for_repository(repoName) if pkg_to_enable
-      next if repositories.find_by_name(repoName)
-
-      # copy target repository when operating on a channel
-      targets = repo.release_targets if (pkg_to_enable && pkg_to_enable.is_channel?)
-      # base is a maintenance incident, take its target instead (kgraft case)
-      targets = repo.release_targets if repo.project.is_maintenance_incident?
-
-      target_repos = []
-      target_repos = targets.map(&:target_repository) if targets
-      # or branch from official release project? release to it ...
-      target_repos = [repo] if repo.project.is_maintenance_release?
-
-      update_project = repo.project.update_instance
-      if update_project != repo.project
-        # building against gold master projects might happen (kgraft), but release
-        # must happen to the right repos in the update project
-        target_repos = Repository.find_by_project_and_path(update_project, repo)
+    if project.is_a? Project
+      # shall we use the repositories from a different project?
+      project = project.update_instance('OBS', 'BranchRepositoriesFromProject')
+      skip_repos = []
+      a = project.find_attribute('OBS', 'BranchSkipRepositories')
+      skip_repos = a.values.map(&:value) if a
+      project.repositories.each do |repo|
+        next if skip_repos.include? repo.name
+        repoName = opts[:extend_names] ? repo.extended_name : repo.name
+        next if repo.is_local_channel?
+        pkg_to_enable.enable_for_repository(repoName) if pkg_to_enable
+        next if repositories.find_by_name(repoName)
+
+        # copy target repository when operating on a channel
+        targets = repo.release_targets if (pkg_to_enable && pkg_to_enable.is_channel?)
+        # base is a maintenance incident, take its target instead (kgraft case)
+        targets = repo.release_targets if repo.project.is_maintenance_incident?
+
+        target_repos = []
+        target_repos = targets.map(&:target_repository) if targets
+        # or branch from official release project? release to it ...
+        target_repos = [repo] if repo.project.is_maintenance_release?
+
+        update_project = repo.project.update_instance
+        if update_project != repo.project
+          # building against gold master projects might happen (kgraft), but release
+          # must happen to the right repos in the update project
+          target_repos = Repository.find_by_project_and_path(update_project, repo)
+        end
+
+        add_repository_with_targets(repoName, repo, target_repos, opts)
```

Comment:
I just intended the code because of this: https://github.com/openSUSE/open-build-service/pull/2932/commits/78f3272aecc36652100b3e3d27677dafef938913#diff-06133ea6068a576c1f21fe39cfdf8536R930